geode-core/src/upgradeTest/java/org/apache/geode/internal/cache/rollingupgrade/RollingUpgradeDUnitTest.java

```diff
@@ -191,20 +207,26 @@ void doTestRollAll(String regionType, String objectType, String startingVersion)
       }
 
       putAndVerify(objectType, server1, regionName, 0, 10, server2);
-      locator = rollLocatorToCurrent(locator, hostName, locatorPorts[0], getTestMethodName(),
-          locatorString);
+      locator =
+          rollLocatorToCurrent(locator, hostName, locatorPort, locatorProps, getTestMethodName(),
+              locatorString);
 
       server1 = rollServerToCurrentAndCreateRegion(server1, regionType, testingDirs[0],
-          shortcutName, regionName, locatorPorts);
+          shortcutName, regionName, new int[] {locatorPort});
       verifyValues(objectType, regionName, 0, 10, server1);
       putAndVerify(objectType, server1, regionName, 5, 15, server2);
       putAndVerify(objectType, server2, regionName, 10, 20, server1);
 
       server2 = rollServerToCurrentAndCreateRegion(server2, regionType, testingDirs[1],
-          shortcutName, regionName, locatorPorts);
+          shortcutName, regionName, ports);
       verifyValues(objectType, regionName, 0, 10, server2);
       putAndVerify(objectType, server2, regionName, 15, 25, server1);
 
+      String shutDownCommand = "shutdown --include-locators=true";
```

Comment:
The shutdown command had also issues. That's why the variable was there initially in the test case.
Could you add the execution of the shutdown command after the ```list members``` in order to verify that with your solution it works?
Otherwise, the variable must be removed.

Reply:
My latest commit eliminates `gfsh` entirely so the issue with `shutDownCommand` is moot.

I eliminated use of `gfsh` because I found the test had become flaky on versions 1.3.0-1.8.0, at least on macOS. On Linux (CI) I did not see the flakiness. But rather than figuring that out, I decided to grab the view more directly: by accessing `InternalDistributedSystem` in the locator JVM.

Reply:
Overall, great work. Thanks, Bill.